src/player_display.cpp

```diff
@@ -1372,14 +1380,21 @@ void player::disp_info()
         draw_stats_tab( w_stats, *this, line, curtab );
     } );
 
+    // TRAITS & BIONICS
+    const std::pair<unsigned int, unsigned int> trait_and_bionic_height =
+        calculate_shared_column_win_height(
+            static_cast<unsigned>( TERMY ) - infooffsetybottom,
+            trait_win_size_y_max,
+            bionics_win_size_y_max
+        );
```

Comment:
This should be inside the `on_screen_resize` callback. `TERMY` is dependent on the game window size so this value needs to be calculated every time the screen is resized.

Reply:
Question: `on_screen_resize` is in lambda catching context. So it is all in one context. Rather than calculating it twice for TRAITS & BIONICS, where can I best put code that works for both but has to be run only once? It would work putted in either `ui_traits.on_screen_resize` or `ui_bionics.on_screen_resize`, but that's disgusting. Is there some universal `ui_adaptor` type on whom I would call `.on_screen_resize(...)` ?

Reply:
I think you can cache the result based on the argument values. Or just calculate it twice. It's not like these are super expensive operations.

Reply:
I will calculate it twice. My motivation was less code duplication, not speed :)